### ISSUE TITLE
Automate the CHANGELOG.md via GH automatically generated release notes.

### ISSUE THREAD

**Issue by sneridagh** [plone/volto#3953]
I'd like to improve the way we are maintaining the CHANGELOG.md in Volto.

At kitconcept, we've been trying lately this approach, with good results:

https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes

Goals:
- the contributor only have to take care of the name of the PR and the categorisation via labels
- the maintainer does not have to take care of merge conflicts 

Also, the new branching policy, and subsequent PRs backporting should benefit from it.

**Comment by tiberiuichim** [comment]
I'm +1, we use something similar at EEA. It's a positive change. I think sometimes there's more context in the PR (description, comments, etc), so it would be nice if we also get links to the PR (I think we'll do, right)?

**Comment by davisagli** [comment]
Yes, the release notes generated by github include a link to the PR and the author of the PR.

**Comment by sneridagh** [comment]
@tiberiuichim it does it, indeed.

It's something like:

```
* Title (#ISSUE_ID) by @AUTHOR in https://github.com/URL/ISSUE
```

**Comment by sneridagh** [comment]
@jensens asking you as label keeper :) 

We have labels marking the PRs as "bug" and "feature". 
- '01 type: bug'
- '03 type: feature (plip)'
- '04 type: enhancement'
 
We'll need a couple more if we want this:
- Breaking
- Documentation
and an explicit "Internal" would be great too.

What do you think?

**Comment by davisagli** [comment]
I do have a concern about adding another way to handle changelogs. It makes it harder for Plone contributors to do the right thing if they work in multiple repositories. Is there a way to use towncrier with a node project?

**Comment by sneridagh** [comment]
One can always execute something in a shell. We'll have to install it locally first in a venv... then a convenience Makefile command, maybe...

**Comment by erral** [comment]
I like the way of prefixing the commits with fix, feat, etc. and render the changelog using auto-changelog red-turtle folks are also using it in some addons like volto-form-block may be @cekk or @pnicolli can also comment.

**Comment by sneridagh** [comment]
So, during the Volto Team meeting we discussed this. We heard feedback from 'commitlint' + release-it from @pnicolli and from kitconcept using GHA automated release notes. We shared opinions from Towncrier too.

The general opinion was that Towncrier could be the best bet on this since
- uses a detached from the flow (in a file) way of define the final message in the changelog
- when backporting this could be a very interesting feature
- is used already in the Plone ecosystem, so contributors do not have to learn another flow
- the downside of creating an issue first to assign it to the news item might be overcame easily, by defining the PR itself or other unique number

as downside, the maintainer(s) need to install Towncrier before the release in an Python env. An automatic way of executing and extracting it when using release-it might be handy... maybe using pipx is a good fit for this...

Since we are already been biten by this when back porting,
